pkg/daemon/daemon.go

```diff
@@ -848,6 +848,11 @@ func (dn *Daemon) isUnspecifiedOS(osImageURL string) bool {
 
 // checkOS validates the OS image URL and returns true if they match.
 func (dn *Daemon) checkOS(osImageURL string) bool {
```

Comment:
Just to clarify, the func returns true if no update and false if it should be updated?

Reply:
ie false = os doesn't match target and we can update, true = os matches or for some other reason we can't update ?

Reply:
if that's the case can we update the comment above the func to make it clear?

Reply:
Yeah, it is a bit odd; we can clean it up but I want to get a quick fix in for the UPI/RHEL folks.

Reply:
makes sense!

Reply:
I would enhance the naming of this function to something like  `systemMatchesOSImageURL` and document that we return false if the system is not support at all to help clear out confusion, not a blocker, let's follow up